src/main/java/org/jitsi/impl/protocol/xmpp/ChatRoomImpl.java

```diff
@@ -225,6 +226,30 @@ public String getMainRoom() {
         return mainRoom;
     }
 
+    /**
+     * Prepare this {@link ChatRoomImpl} for a call to {@link #joinAs(Resourcepart)}, which send initial presence to
+     * the MUC. Resets any state that might have been set the previous time the MUC was joined.
+     */
+    private void resetState()
```

Comment:
I thought that ChatRoomImpl instances are destroyed and a new one would be created? Is there a rejoin scenario?

Reply:
They are cached and potentially reused in `XmppProtocolProvider.Muc`. I'm not sure it's necessary, but I don't understand it well enough to change right now.

This fix is based on a dump where a chat room was re-joined and had an old entry in `members` which was never removed as it is not an occupant of the Smack MUC.